### PR TITLE
Compatibility with `term` *0.2.9*

### PR DESCRIPTION
**Pull request by Byron**
When attempting to compile cargo with the latest `term` crate,
one would face compilation issues due to an attempt to declare
a `Terminal` type without its apparently new associated type
called `Output`.

The fix involves the removal of the `UghWHyIsThisNecessary` type,
and declares the `Terminal` type as
`Terminal<Output=Box<Write + Send>>`.

**Note:** The reason I ran into this was an accidental update to the `Cargo.lock` file, which pulled in the said version of `term`.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Can this updated just the `term` dependency? I'd prefer to hold off on all the other updates for now.

**Comment by Byron** [comment]
I also think it's a good idea to keep the change as focussed as possible, thanks for the reminder.

**Comment by Byron** [comment]
Sorry for not adhering to the contribution guidelines, as I force-pushed the fixes which makes reviews harder in all but the trivial cases.

File: src/cargo/core/shell.rs [hunk 1]
```diff
@@ -185,13 +180,4 @@ impl Write for Shell {
...
-
-impl Write for UghWhyIsThisNecessary {
-    fn write(&mut self, bytes: &[u8]) -> io::Result<usize> {
-        self.inner.write(bytes)
-    }
-    fn flush(&mut self) -> io::Result<()> {
-        self.inner.flush()
-    }
-}
+}
```
Comment (globin) [inline src/cargo/core/shell.rs]: Missing new line at EOF

**Comment by alexcrichton** [comment]
@bors: r+ 3b9753b539863dbd68d1702970d33fdab6f86992

Thanks!

**Pull request merged**